go/cmd/dolt/commands/merge.go

```diff
@@ -345,54 +392,68 @@ func calculateMergeConflicts(queryist cli.Queryist, sqlCtx *sql.Context, mergeSt
 		}
 	}
 
-	return mergeStats, dataConflicts == nil && schemaConflicts == nil && constraintViolations == nil, 0
+	return mergeStats, dataConflicts == nil && schemaConflicts == nil && constraintViolations == nil, true
 }
 
-// calculateMergeStats calculates the table operations and row operations that occurred during the merge.
-func calculateMergeStats(queryist cli.Queryist, sqlCtx *sql.Context, mergeStats map[string]*merge.MergeStats) (map[string]*merge.MergeStats, int) {
-	diffSummaries, err := getDiffSummariesBetweenRefs(queryist, sqlCtx, "head^1", "head")
+// calculateMergeStats calculates the table operations and row operations that occurred during the merge. Retyrns a map of
+// table name to MergeStats, and a bool indicating whether calculation was successful.
+func calculateMergeStats(queryist cli.Queryist, sqlCtx *sql.Context, mergeStats map[string]*merge.MergeStats, fromRef string) (map[string]*merge.MergeStats, bool) {
+	diffSummaries, err := getDiffSummariesBetweenRefs(queryist, sqlCtx, fromRef, "HEAD")
 	if err != nil {
+		cli.Println("merge successful, but could not calculate stats")
 		cli.Println(err.Error())
-		return nil, 1
+		return nil, false
 	}
 
 	diffStats := make(map[string]diffStatistics)
 
+	var allUnmodified = true
 	// get table operations
 	for _, summary := range diffSummaries {
 		if summary.DiffType == "added" {
+			allUnmodified = false
 			mergeStats[summary.TableName] = &merge.MergeStats{
 				Operation: merge.TableAdded,
 			}
 		} else if summary.DiffType == "dropped" {
+			allUnmodified = false
 			mergeStats[summary.TableName] = &merge.MergeStats{
 				Operation: merge.TableRemoved,
 			}
 		} else if summary.DiffType == "modified" || summary.DiffType == "renamed" {
+			allUnmodified = false
 			mergeStats[summary.TableName] = &merge.MergeStats{
 				Operation: merge.TableModified,
 			}
-			tableStats, err := getTableDiffStats(queryist, sqlCtx, summary.TableName, "head^1", "head")
+			tableStats, err := getTableDiffStats(queryist, sqlCtx, summary.TableName, fromRef, "HEAD")
 			if err != nil {
+				cli.Println("merge successful, but could not calculate stats")
 				cli.Println(err.Error())
-				return nil, 1
+				return nil, false
+			}
+			if tableStats != nil && len(tableStats) > 0 {
+				diffStats[tableStats[0].TableName] = tableStats[0]
 			}
-			diffStats[tableStats[0].TableName] = tableStats[0]
 		} else {
 			mergeStats[summary.TableName] = &merge.MergeStats{
 				Operation: merge.TableUnmodified,
 			}
 		}
 	}
 
+	if allUnmodified {
+		cli.Println("Already up to date.")
```

Comment:
moving all cli.Println out of the `calculateMergeStats` function would be ideal, but may be impractical. See what you can do.